xgmutil/stream.py

```diff
@@ -1,26 +1,33 @@
 import jax.numpy as jnp 
 import jax.random as rnd
 import jax
-import sys
 import os
 
 class Stream:
     '''Stream'''
     def __init__(self, **kwargs):
 
         self.force_no_gpu = kwargs.get('force_no_gpu',False)
-        self.seed         = kwargs.get('seed',123456789)
+        self.seedkey      = kwargs.get('seedkey', rnd.PRNGKey(123456789))
         self.nsub         = kwargs.get('nsub',1024**3)
+        self.dtype        = kwargs.get('dtype', jnp.float32)
 
     def generate(self,**kwargs):
 
         if self.force_no_gpu:
-            _JAX_PLATFORM_NAME = os.environ["JAX_PLATFORM_NAME"]
-            os.environ["JAX_PLATFORM_NAME"] = "cpu"
+            _JAX_PLATFORM_NAME = jax.default_backend()
+            jax.default_device("cpu")
 
         start = kwargs.get('start',0)
         size  = kwargs.get('size' ,1)
         dist  = kwargs.get('dist','normal')
+        mc    = kwargs.get('mc', 0)
```

Comment:
If I understand correctly, to retain the previous functionality as an option (i.e. user provides an integer seed instead of a key), then the `mc` variable would assumed to be zero?

Reply:
I presume if we have a `seed` key, the `mc` key can be degenerate depending on how it is used. One can generate new realizations by changing the seed (or the PRNGKey) itself and creating a new stream. But I would advice against that. The intended used would be to initialize once with a seed or PRNGKey and then independent realizations by changing `mc`. I think by default `mc` is always assumed to be zero, whether we use PRNGKey or seed. I hope that answers your question.

Reply:
Thanks for the explanation. I now understand that `mc` is an integer that uniquely labels a given realization, which is folded into a given PRNGKey. As is, if `self.seedkey` is an integer, then the value of `mc` will have no effect, preserving the existing functionality in main that allows the more straightforward approach of providing an integer seed without having to pass in a key.